tests/system/providers/amazon/aws/utils/__init__.py

```diff
@@ -92,15 +93,15 @@ def _fetch_from_ssm(key: str, test_name: str | None = None) -> str:
     :return: The value of the provided key from SSM
     """
     _test_name: str = test_name if test_name else _get_test_name()
-    ssm_client: BaseClient = boto3.client("ssm")
+    hook = SsmHook()
```

Comment:
Hey @ferruzzi I guess the issues with tests here.
Due to the nature of AwsBaseHook by default it use `aws_conn_id=aws_default` as result when you try to create SSM Client it Hook tried to obtain connection first.

I think if you set `aws_conn_id` to `None` it should not use Airflow Connection

```suggestion
    hook = SsmHook(aws_conn_id=None)
```

Reply:
I had two solutions I was playing with and that was not one of them.  I'll try that out, if it works it would be a much less intrusive fix than the one I decided to implement.

Reply:
Thinking this through, that would mean that the connection itself is not being tested.  Is that something we want to accept in a system test?

I think it's alright in this case.  The system tests are supposed to be testing the specific services, not whether the connection to SSM is working.... but that does mean it's a less-complete system test.

Reply:
> Thinking this through, that would mean that the connection itself is not being tested.

We tests connections for AwsBaseHook and all other helpers. Actually there is 3 cases and all of them tested:
1. aws_conn_id = None
2. aws_conn_id = not-existed-conn, we know only after we tried connect to DB (one in the future we would remove this one)
3. aws_conn_id = exists-conn

So we don't need test for all Hooks which  based on AwsBaseHook and AwsGenericHook and how they obtain connection from Airflow.

The actual error here happen in this test

https://github.com/apache/airflow/blob/877189916900c930b2c4b92101d46d2f98eb9077/tests/always/test_example_dags.py#L62-L68

I'm not familiar with this test, but seems like it test that example DAGs do not perform queries to Airflow Database during import example dags: https://github.com/apache/airflow/pull/7419 

I don't know maybe this test less relevant nowadays

Reply:
Yeah, I was digging in that code yesterday and one option I worked on was adding AWS system tests as an exception and to assert that AWS tests make exactly one query, but I don't really know how important that restriction is. so I was trying to look into the restriction and why it's there.  It looks like your change does fix the test (this PR is passing now) so maybe let's run with this.

@o-nikolas, can you have a look at this when you get a few minutes and let me know what you think?  I think I'm inclined to agree with Taragolis that this is the right solution here.

Reply:
I don't love changing the "prod" code in this way to make the tests pass (I think the expectations in that unit test should be updated for AWS examples). But on the other hand the "prod" code here is technically also test code, so it's not so bad and it's a clean change :+1: Happy to merge!